### PR TITLE
PKX-1437

### PR DESCRIPTION
**Pull request by hamzatanveer012**
## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/extensions/payment/processors/xstack.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        amount = basket.total_incl_tax
+        user_name = basket.owner.username
+        data = self.create_ordered_dict([
+            ('customerName', user_name),
+            ('amount', amount),
+            ('orderRefNum', order_id),
+        ])
+        logger.info('Generated data: {}'.format(data))
+        query_str = urlencode(data)
+        logger.info('Generated str: {}'.format(query_str))
```
Comment (m-usman01) [inline ecommerce/extensions/payment/processors/xstack.py:79]: Too many logs, add a TODO comment here to remove these once everything has been finalised

File: ecommerce/extensions/payment/views/xstack.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                ('name', "jjj"),
+                ('phone', "876876"),
+            ])),
+            ('shipping', OrderedDict([
+                ('address1', "kjhkj"),
+                ('city', "kjhkj"),
+                ('country', "kjhkj"),
+                ('province', "79"),
+                ('zip', "987")
+            ])),
```
Comment (m-usman01) [inline ecommerce/extensions/payment/views/xstack.py]: Why are we hard coding this?

File: ecommerce/settings/base.py
Comment (ahmadshafique) [inline ecommerce/settings/base.py:1]: Unnecessary changes. Pls remove them

**Review by m-usman01 — APPROVED** [review]

File: ecommerce/settings/base.py [hunk 3]
```diff
@@ -779,11 +784,6 @@
         }
     }
 }
-MEDIA_STORAGE_BACKEND = {
```
Comment (ahmadshafique) [inline ecommerce/settings/base.py]: Please revert these changes as well

Comment (ahmadshafique) [reply ecommerce/settings/base.py]: @hamzatanveer012 ^

File: ecommerce/extensions/payment/views/xstack.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            ('payment_method_types', "card"),
+            ('customer', OrderedDict([
+                ('email', data.data['email']),
+                ('name', '{} {}'.format(data.data['first_name'], data.data['last_name'])),
+                ('phone', data.data['phone_number']),
+            ])),
+            ('shipping', OrderedDict([
+                ('address1', '{}, {}'.format(data.data['street_address'], data.data['address_line2'])),
+                ('city', data.data['city']),
+                ('country', ''),
```
Comment (ahmadshafique) [inline ecommerce/extensions/payment/views/xstack.py]: @hamzatanveer012 pass country and province data as passed from the updated form by Tanveer.

**Review by ahmadshafique — CHANGES_REQUESTED** [review]

**Review by ahmadshafique — APPROVED** [review]

**Pull request merged**